### PR TITLE
HSEARCH-3603 Swap the names of BeanResolver and BeanProvider

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-3603
update: // -> /

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/engine/environment/bean/BeanResolver.java [hunk 1]
```diff
@@ -11,74 +11,77 @@
...
+ * {@link #resolve(Class) by their type},
+ * or {@link #resolve(Class, String) by their type and name},
+ * or {@link #resolveRole(Class) by their role}.
  * <p>
- * This interface may be used by any Hibernate Search module,
- * but should only be implemented by the Hibernate Search engine itself;
- * if you are looking for implementing your own bean resolver,
- * you should implement {@link BeanResolver} instead.
+ * This interface is API,
+ * but should only be implemented by Hibernate Search itself;
```
Comment (fax4ever) [inline engine/src/main/java/org/hibernate/search/engine/environment/bean/BeanResolver.java]: Then maybe we should move it from a spi to an impl package.

Comment (yrodiere) [reply engine/src/main/java/org/hibernate/search/engine/environment/bean/BeanResolver.java]: I was trying to say that even though users can *call* the methods of this interface, nobody should not try to *implement* this interface. If people want to customize the bean retrieval behavior, they should implement `BeanProvider` instead (as explained below).

**Comment by yrodiere** [comment]
Rebased, waiting for CI

**Comment by yrodiere** [comment]
Merged, thanks!

**Pull request merged**